solar_system.rb

```diff
@@ -0,0 +1,211 @@
+  learn_more = y_or_n(learn_more)
+  if learn_more == "Y"
+    continue ="Y"
+    while continue == "Y"
+      puts "Please write the planet's name."
+      user_interest = gets.chomp.downcase
+
+      while !input_is_name(my_solar_system, user_interest)
+        puts "That is not an option.  Please select a planet on the list."
+        user_interest = gets.chomp.downcase
+      end
+
+      puts "\nHere is information you asked for:"
+      n=0
+      my_solar_system.planets.length.times do
+        if my_solar_system.planets[n].name.downcase == user_interest
+          output = my_solar_system.planets[n].planet_all_info
+          puts output
+        end
+        n +=1
+      end
+
+      puts "\n Would you like to hear about another planet? (Y/N)"
+      continue = gets.chomp.upcase
+      continue = y_or_n(continue)
+    end
+  else
+    puts "Ok, I won't tell you any more about my planets"
+  end
+end
+
+#prompts the user to input all the the information to define a new planet
+#and makes the new planet
+def make_a_planet(name)
+  puts "How long is a year on #{name}? (in earth-days)"
+  year_length = gets.chomp
+  puts "How far form the sun is #{name} (in units)?"
+  distance_from_sun = gets.chomp
+  puts "If this was a star trek episode, what type of planet is #{name}?"
+  type = gets.chomp
+  puts "How many moons does #{name} have?"
+  moons = gets.chomp
+  puts "Does an intelligent species live on #{name}? (Y/N)"
+  species = gets.chomp.upcase
+  species = y_or_n(species)
+  if species == "N"
+    inhabitants = "No inteligent life"
+  else
+    puts "What is the species name?"
+    inhabitants = gets.chomp
+  end
+  new_planet = Planet.new(name, year_length, distance_from_sun, type, moons, inhabitants)
+  return new_planet
+end
+
+#asks the user if they would like to define a new planet.
+# If yes, it allows them too.
+def user_defined_planet(my_solar_system)
+  puts "I can help you define a new planet. Would you like to?"
+  wants_to_define = gets.chomp.upcase
+  wants_to_define = y_or_n(wants_to_define)
+
+  if wants_to_define == "Y"
+    puts "Great! Please enter the name of your new planet"
+    name = gets.chomp.capitalize
+    new_planet = make_a_planet(name)
+    puts "This is the information on the planet you just entered"
+    puts new_planet.planet_all_info
+    #could ask user if that info is correct here, and allow changes if not.
+    my_solar_system.add_planet(new_planet)
+    puts "\nHere is the updated list of planets in my solar system"
+    puts my_solar_system.ss_name_list
+    puts ""
+  else
+    puts "Well OK then."
+  end
+  return my_solar_system
+end
+
+#SETTING VARIABLES
+#these are hardcoded planets in my starter solar system
+first = Planet.new("Caprica", 102, 230, "M", 4, "Humans")
+second = Planet.new("Vulcan", 40, 100 ,"M", 2, "Vulcans")
+third = Planet.new("Delta Vega", 50, 1100, "A", 1, "No inteligent life" )
+forth = Planet.new("Alpha Onias III", 102, 230, "M", 1, "Barash")
```

Comment:
Since you're creating all these local variables just to add them to an array, a more concise way to do this might be:

```ruby
planet_array = [
  Planet.new("Caprica", 102, 230, "M", 4, "Humans"),
  Planet.new("Vulcan", 40, 100 ,"M", 2, "Vulcans"),
  # ...
]
my_solar_system = SolarSystem.new(planet_array)
```

This makes it clear to the reader that we won't be using the planets individually, only through the collection, and also saves you having to come up with a bunch of variable names.